capstone/src/test/java/com/google/sps/CachedInterestVectorTest.java

```diff
@@ -0,0 +1,148 @@
+package com.google.sps;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.Room;
+import com.google.sps.data.User;
+import com.google.sps.data.Tag;
+import com.google.sps.data.CachedInterestVector;
+import com.google.sps.service.DatabaseService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+import java.lang.Math;
+
+@RunWith(JUnit4.class)
+public final class CachedInterestVectorTest {
+
+    private final LocalServiceTestHelper helper =
+        new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    private static final double epsilon = .00001;
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    @Test
+    public void embeddedEntityToHashMapTest(){
+        Double magnitude = 3.0;
+        EmbeddedEntity embeddedVectorMap = new EmbeddedEntity();
+        embeddedVectorMap.setProperty(Tag.EDUCATION.getTag(), (2.0/magnitude));
+        embeddedVectorMap.setProperty(Tag.COOKING.getTag(), (2.0/magnitude));
+        embeddedVectorMap.setProperty(Tag.FITNESS.getTag(), (1.0/magnitude));
+
+        HashMap<String, Double> vectorHashMap = CachedInterestVector.embeddedEntityToHashMap(embeddedVectorMap);
+        Assert.assertEquals(vectorHashMap.get(Tag.EDUCATION.getTag()), (2.0/magnitude), epsilon);
+        Assert.assertEquals(vectorHashMap.get(Tag.COOKING.getTag()), (2.0/magnitude), epsilon);
+        Assert.assertEquals(vectorHashMap.get(Tag.FITNESS.getTag()), (1.0/magnitude), epsilon);
+    }
```

Comment:
Add to the test (or a separate test if you want) what happens when you explicitly put null as the value for a property. What about zero? What about a negative number? Since this is simply a conversion method to swap entities and hash maps, Im not necessarily saying error detection needs to happen here (even if negative numbers are invalid, then it might not throw and might just have a negative in the hashmap since validation is not the goal of this method: thats fine, but it should be demonstrated in the test)

Similar cases for the reverse conversion method test below